src/arrow/array/fixed_size_list.rs

```diff
@@ -277,16 +277,21 @@ mod tests {
             .collect::<FixedSizeListArray<2, StringArray, true>>();
         let fixed_size_list_array_nullable =
             arrow_array::FixedSizeListArray::from(fixed_size_list_array_nullable_input);
+
+        let owned_input_nullable = INPUT_NULLABLE
+            .into_iter()
+            .map(|item| item.map(|[first, second]| [first.to_owned(), second.to_owned()]))
+            .collect::<Vec<_>>();
         assert_eq!(
             FixedSizeListArray::<
                 2,
                 StringArray<false, i32, crate::arrow::buffer::ScalarBuffer>,
                 true,
-                crate::arrow::buffer::ScalarBuffer,
```

Comment:
Seems like `arrow_buffer::ScalarBuffer<T>` does not implement `IntoIterator` for owned values.

Reply:
This, for example, also does not compile:
```rust
fn into_iter() {
    let input = [Some(1), Some(2), Some(3), Some(4), None, Some(42)];
    let nullable = input
        .into_iter()
        .collect::<Nullable<Vec<_>, crate::arrow::buffer::ScalarBuffer>>();
    let output = nullable.into_iter().collect::<Vec<_>>();
    assert_eq!(input, output.as_slice());
}
```

Reply:
We could try to provide that impl upstream?

Reply:
gonna punt this for now.